challonge/api.py

```diff
@@ -118,17 +106,44 @@ def _prepare_params(dirty_params, prefix=None):
     objects.
 
     """
-    params = {}
-    for k, v in dirty_params.items():
-        if hasattr(v, "isoformat"):
-            v = v.isoformat()
-        elif isinstance(v, bool):
-            # challonge.com only accepts lowercase true/false
-            v = str(v).lower()
-
-        if prefix:
-            params["%s[%s]" % (prefix, k)] = v
+    if prefix and prefix.endswith('[]'):
+        keys = []
+        values = []
+        for k, v in dirty_params.items():
+            if isinstance(v, (tuple, list)):
+                keys.append(k)
+                values.append(v)
+        firstiter = ((k, v) for vals in zip(*values) for k, v in zip(keys, vals))
+        lastiter = ((k, v) for k, v in dirty_params.items() if k not in keys)
+        dpiter = itertools.chain(firstiter, lastiter)
+    else:
+        dpiter = dirty_params.items()
+
+    params = []
+    for k, v in dpiter:
+        if isinstance(v, (tuple, list)):
+            for val in v:
+                val = _prepare_value(val)
+                if prefix:
+                    params.append(("%s[][%s]" % (prefix, k), val))
```

Comment:
Are you sure that the string here should be `"%s[][%s]"`? Because I have it as `"%s[%s][]"` and it has been working for me (last tested on January 20, 2016). Has it changed since then?